github-extension/background.js

```diff
@@ -0,0 +1,113 @@
+      for (const {node: repo} of contributedRepositories.edges) {
+        const {nameWithOwner, pullRequests, issues} = repo;
+        for (const {node: pr} of pullRequests.edges) {
+          const {author, number, title, assignees, url, reviewRequests, updatedAt} = pr;
+          const isAuthor = author.login === login;
+          const isAssigned = assignees.edges.some(({node: assignee}) => assignee.login === login);
+          const isReviewRequested = reviewRequests.edges.some(({node: rr}) => rr.reviewer.login === login);
+          if (isAuthor || isAssigned || isReviewRequested) {
+            let significance;
+            if (isReviewRequested) {
+              significance = "Your review requested";
+            } else if (isAssigned) {
+              significance = "Assigned to you";
+            } else {
+              significance = "You created";
+            }
+            priorityCards.push({
+              title: `PR #${number}: ${significance}`,
+              description: title,
+              url,
+              image: author.avatarUrl,
+              hostname: nameWithOwner,
+              updatedAt
+            });
+          }
+        }
+        if (issues) {
+          for (const {node: issue} of issues.edges) {
+            const {author, number, title, assignees, url, updatedAt} = issue;
+            const isAuthor = author.login === login;
+            const isAssigned = assignees.edges.some(({node: assignee}) => assignee.login === login);
+            if (isAuthor || isAssigned) {
+              let significance;
+              if (isAssigned) {
+                significance = "Assigned to you";
+              } else {
+                significance = "You reported";
+              }
+              priorityCards.push({
+                title: `Issue #${number}: ${significance}`,
+                description: title,
+                url,
+                image: author.avatarUrl,
+                hostname: nameWithOwner
+              });
+            } else {
+              fillerCards.push({
+                title: `Issue #${number}`,
+                description: title,
+                url,
+                image: author.avatarUrl,
+                hostname: nameWithOwner,
+                updatedAt
+              });
+            }
+          }
+        }
+      }
+      priorityCards.sort((a, b) => Date.parse(b.updatedAt) - Date.parse(a.updatedAt));
+      fillerCards.sort((a, b) => Date.parse(b.updatedAt) - Date.parse(a.updatedAt));
+      const allCards = priorityCards.concat(fillerCards);
+      allCards.forEach(card => delete card.updatedAt);
+      resolve(allCards);
+    });
+    req.open("POST", apiUrl);
+    req.setRequestHeader("Content-type", "application/json");
+    req.send(requestBody);
+  });
+
+const update = async (forceUpdate = false) => {
+  if (!cards || Date.now() - lastUpdated > UPDATE_INTERVAL) {
+    cards = await getCards();
+    browser.newTabSection.addCards(cards, true);
```

Comment:
I'm assuming ```browser``` is some global?

Reply:
Yes it is, it's the entry point for webextension apis